### DISCUSSION TITLE
Docker image creates empty volume on start

### DISCUSSION THREAD

**Discussion by jeffrson** [verdaccio/verdaccio#3000] [Ideas]
**Your Environment**
  
 * **verdaccio version**: 5.5.2
 * **platform**: [npm, docker, helm, other] docker

**Describe the bug**

On each `docker run` (also via docker compose) a *new* volume for /verdaccio/storage is created. IMO, it should be left to the user, whether and how the volume should be created at all. Besides, the empty volume on start is rather useless.
For me, I'd like to use a bind mount from a certain host directory to /verdaccio. However, at the moment, I need to override the volume at /verdaccio/storage in order to prevent those <docker volume id> volumes and, most notably, reuse all uploaded packages.

**To Reproduce**

```
docker run --rm -d verdaccio/verdaccio:5.5.2
docker volume ls
```

**Expected behavior**

Doesn't create any unexpected docker volume.

**Comment by juanpicado** [comment]
Works for me
```
docker --version
Docker version 20.10.12, build e91ed57

➜  docker run --rm -d verdaccio/verdaccio:5.5.2
543dc................................................5474
➜  docker volume ls                            
DRIVER    VOLUME NAME
local     .708b2e00f.................0b324e8............
```

**Comment by jeffrson** [comment]
Yes, and with next `docker run...` it will create another volume a.s.o., always forgetting about the previous.

**Comment by juanpicado** [comment]
> IMO, it should be left to the user, whether and how the volume should be created at all. 
> For me, I'd like to use a bind mount from a certain host directory to /verdaccio

Ok I see your point,I don't see the big issue, but anyway, do you have any specific proposal that improves what exist already?

**Comment by jeffrson** [comment]
The issue is "simple": after restarting verdaccio, all locally published npm packages are gone. They went away with the old inaccessible volume. All those private package will need to be republished.

Suggestion would be to remove volume definition from Dockerfile, since anonymous volumes are not very useful these days anyway. Everybody would be free to define a volume as they like, be it for /verdaccio or /verdaccio/storage, /verdaccio/config, /verdaccio/plugins.

**Comment by juanpicado** [comment]
I understand but I don't agree, I think are very useful, volumes can be mapped if you want to persist you have volumes, docker-compose, etc etc, anyhow I guess is just matter of opinions, I respect your of course. 

This is not a bug, rather a suggestion so I'll move this thread to a disscusion and tag the Helm/Kubernetters users which have more experience than me in this matter and keep discussing there. cc: @verdaccio/kubernetes

**Reply by wakecoder** [reply]
I don't see any disadvantage to removing the volume from the Dockerfile as it's very easy to create a volume using Docker, docker-compose or K8s. 

There are instances where it's useful to not have a volume and docker commit after doing an npm install pointing to a verdaccio container. This allows you to create an image that includes all the packages that were pulled. This is useful if you want to provide an npm registry with specific packages in an offline environment as you can just move the image to your offline environment.

With the volume specified in the Dockerfile you can't do this without manually editing the dockerfile and rebuilding the image.

**Comment by jeffrson** [comment]
Sure, YMMV :-)

I'm really interested in the benefit of anonymous volumes nowadays. Is there more than "--volume-from"?

**Comment by joaomelorodrigues** [comment]
I'm having this issue as well. I mapped the volume storage outside the container and it simply doesn't work on publish new npm packages. However it works if I don't have the volume mapped, but then every docker image update will wipe all the published packages. This is really anoying.

**Comment by tu-andy** [comment]
Why is this default behavior? Literally no other docker image I have used has this issue
